doc/ipythonconsole.rst

```diff
@@ -2,91 +2,159 @@
-   * If using a version of Spyder before 3.3.0, ``ipython kernel`` should work to launch the kernel, albeit without certain Spyder-specific features.
+#. In Spyder, click :guilabel:`Connect to an existing kernel` from the :guilabel:`Consoles` menu, and paste the name of the :guilabel:`Connection file` from the previous step.
 
-#. Copy the connection file (:file:`{jupyter/runtime/dir/path}/kernel-{pid}.json`) to the machine you're running Spyder on (if remote) or note its location (if local).
+   As a convenience, kernel ID numbers (e.g. ``1234``) entered in the connection file path field will be expanded to the full path of the file, i.e. :file:`{jupyter/runtime/dir/path}/kernal-{id}.json`.
 
-   You can get :file:`{jupyter/runtime/dir/path}` by executing ``jupyter --runtime-dir`` in the same Python environment as the kernel.
+   .. image:: images/console/console-connect-local-step2.gif
+      :alt: Copying the connection filename into Spyder's dialog
 
-#. Click :guilabel:`Connect to an existing kernel` from the :guilabel:`Console` menu or the :guilabel:`IPython Console` pane's "Gear" menu.
+#. Click :guilabel:`OK` to connect to the kernel.
 
-#. Browse for or enter the path to the connection file from the previous step.
-   If you're connecting to a local kernel, click :guilabel:`Ok` and Spyder should connect to the kernel; if a remote kernel, proceed to the final step.
+   .. image:: images/console/console-connect-local-step3.gif
+      :alt: Connecting to the kernel and running basic commands.
 
-   As a convenience, kernel ID numbers (e.g. ``1234``) entered in the connection file path field will be expanded to :file:`{jupyter/runtime/dir/path}/kernal-{id}.json` on your local machine.
 
-#. If connecting to a remote kernel over ``ssh``, check the appropriate box and type the full hostname you're connecting to (in the form :file:`{username}@{hostname}:{port-number}`).
-   Then, enter *either* :file:`{username}` 's password on the remote machine, or your user SSH keyfile (typically :file:`.perm`) (only one is needed to connect), and press :guilabel:`Ok`.
+Connect to a remote kernel
+~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-   The port number is the one on which the SSH daemon (``sshd``) is running, typically 22 unless you or your administrator has configured it otherwise.
+To connect to a kernel on a remote machine,
 
-.. image:: images/console/console-dialog-connect.png
-   :alt: Connect to kernel dialog, requesting path and connection details
+#. Launch a Spyder kernel on the remote host if one is not already running, with ``python -m spyder_kernels.console``.
 
-For more technical details about connecting to remote IPython kernels, see the `Connecting to a remote kernel`_ page in the IPython Cookbook.
-Just remember to enter the appropriate details into Spyder's :guilabel:`Connect to an existing kernel` dialog instead of launching a new frontend on the client with ``--existing``.
+   .. image:: images/console/console-connect-remote-step1.gif
```

Comment:
I suggest to run `conda activate spyderenv` instead of `source /~spyderenv` because that has been the default way to activate envs for a couple of years, I think.

The rest looks good to me.

Reply:
Besides, what do you think about also using `nohup` to start the kernel? That way you could close your remote session, or reuse it for the gif below (instead of using two sessions, as you're doing below).

Reply:
> I suggest to run `conda activate spyderenv` instead of `source /~spyderenv` because that has been the default way to activate envs for a couple of years, I think.

I'm not using `conda` at all, its standard `venv`, since its on a Raspberry Pi running Raspbian.

> Besides, what do you think about also using `nohup` to start the kernel? That way you could close your remote session, or reuse it for the gif below (instead of using two sessions, as you're doing below).

I could, but it adds more complexity in exchange for a modest efficiency gain, so I don't know if its worth it. It also makes it more complicated for the user to terminate the kernel later, and might make it less clear which host is which.

Reply:
I've resolved the portions of this that are applicable and run the two in two clearly separate consoles on the same screen, so this should be fixed.